demos/real_time_stream_analysis/python/inference_manager.py

```diff
@@ -0,0 +1,148 @@
+#
+# Copyright (c) 2022 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import queue
+from typing import Tuple
+import numpy as np
+import threading
+import time
+import multiprocessing
+from logger import get_logger
+from inference_executor import InferenceExecutor
+from inference_pipeline_handler import InferencePipelineHandler
+
+
+class InferenceManager:
+
+	def __init__(self, ovms_url, model_name, model_version=0, num_inference_executors=16, binary_input=False,
+				 buffer_size=1000):
+		
+		self.exit_event = threading.Event()
+		self.abort_event = threading.Event()
+
+		self.logger = get_logger(__name__)
+		self.logger.info("Initializing Inference Manager...")
+
+		model_version_str = "latest" if model_version == 0 else model_version
+		self.logger.info(f"OVMS Endpoint spec - ovms_url:{ovms_url}; model_name: {model_name}; model_version: {model_version_str}")
+
+		ovms_info = {
+			"ovms_url": ovms_url, 
+			"model_name": model_name,
+			"model_version": model_version
+		}
+
+		self.logger.info(f"Input buffer capacity set to: {buffer_size} frames")
+		self.inputs_queue = queue.Queue(maxsize=buffer_size)
+		self.results_queue = queue.Queue(maxsize=buffer_size)
+
+		if binary_input:
+			self.logger.info("Using binary input switched on")
+		
+		self.logger.info(f"Number of Inference Executors: {num_inference_executors}")
+		self.logger.info("Starting Inference Executors...")
+		self.inference_executors = [InferenceExecutor(i, ovms_info, binary_input,
+								   input_queue=multiprocessing.Queue(buffer_size),
+								   result_queue=multiprocessing.Queue(buffer_size),
+								   abort_event = multiprocessing.Event())
+								   for i in range(num_inference_executors)]
```

Comment:
is there a reason for creating abort_event in here? Maybe create it inside InferenceExecutor constructor